Add docker makefile

gtonye opened the pull request:
## Highlight

* Dockerize the application
* Package with a makefile

## Details

* Added Debian based Docker to use the app in a container.
* Update requests version to address security notice from Github.
* Update readme to reflect changes.

connected-gtonye committed:
Split scrapping function in two functions
Add support for arguments on the command line
Move code to Python 3+
Add docstring in functions

connected-gtonye committed:
Fix following security alert on requests versions below 2.20

connected-gtonye committed:
Add dockerfile

connected-gtonye committed:
Add Makefile

connected-gtonye committed:
Ignore intelliJ

connected-gtonye committed:
Update the documentation to reflect the changes

odairay commented on .gitignore:
```diff
@@ -2,3 +2,4 @@
 /**/.DS_Store
 /**/*.log
 /.env
+/**/.idea
```
ça devrait plutôt être présent dans ton ignore global, pas dans l'ignore du projet vu que ça concerne ton IDE

gtonye replied:
c'est le gitignore principal du projet.

odairay replied:
non ce que je voulais dire c'est que ça doit être dans ton ignore local à ta machine, tu ne peux pas mettre les fichiers de tous les IDE dans ton ignore, c'est à chaque dev de le gérer de son côté

odairay commented on count_amazon_skills.py:
```diff
@@ -11,6 +13,7 @@
+from selenium.webdriver.firefox.options import Options
 
 
 AMAZON_BASE_URL = 'https://www.amazon.com'
```
plutot à mettre dans un fichier de conf ? ou dans l'environnement ?

odairay commented on Dockerfile:
```diff
@@ -0,0 +1,19 @@
+COPY requirements.txt ./
+RUN pip3 install --no-cache-dir -r requirements.txt
+
+COPY count_amazon_skills.py Makefile ./
```
Makefile ne sert pas dans le container

odairay commented on count_amazon_skills.py:
```diff
@@ -27,56 +30,141 @@
+# when opening or closing a tab require the waiting time below to not trigger
+# errors such as https://stackoverflow.com/questions/27775759
+# /send-keys-control-click-in-selenium-with-python-bindings
+BROWSER_ACTION_WAIT_TIME = 3
```
pareil

odairay commented on count_amazon_skills.py:
```diff
@@ -27,56 +30,141 @@
 
     driver_skill_cat_els = driver.find_elements(By.XPATH, SKILL_CAT_XPATH)
     for skill_cat_el in driver_skill_cat_els:
         skill_cat_name = skill_cat_el.text
```
Variable inutile

gtonye replied:
elles sert un peu plus bas L71

odairay replied:
yes mais du coup tu peux utiliser direct skill_cat_el.text

gtonye replied:
En fait non parce que l'element est modifié par la ligne 91!

odairay replied:
:+1:

odairay commented on count_amazon_skills.py:
```diff
@@ -27,56 +30,141 @@
     return skills
 
 
+def get_no_skill_in_category(driver, skill_cat_el):
```
get_nb_skill_in_category

odairay commented on count_amazon_skills.py:
```diff
@@ -27,56 +30,141 @@
+        if not match_results:
+            match_results = match('([0-9][0-9,.]*) result.*', result.text)
+        if match_results:
+            skill_cat_cnt = match_results.groups()[0]
```
Utilise une autre variable

gtonye replied:
Pourquoi?

connected-gtonye committed:
Add default environment variable for browser type and report type
Add target for build and run and for run only
Support optional environment file for the docker run command

connected-gtonye committed:
Add encoding
Split configuration in a separate python script file
Change method name to be more explicit
Add new variable name for the string counter and the int counter

connected-gtonye committed:
Update Dockerfile to reflect changes in Makefile and with the new config

connected-gtonye committed:
Update readme to include new run commands and environment configuration

connected-gtonye committed:
Update .gitignore to specify what is ignored.

connected-gtonye committed:
rename function to return category and the count of skills in the given category.

odairay commented on config.py:
```diff
@@ -0,0 +1,13 @@
+# *--Encoding: utf-8 --*
+"""Configuration file."""
+
+AMAZON_BASE_URL = 'https://www.amazon.com'
```
Tu peux mettre tes autres constantes dedans du coup et renommer le fichier

connected-gtonye committed:
Change the name of the config file to constants

connected-gtonye committed:
Update docker file and main script to include constants and move variable to constants.
